File: Ahorn/entities/oneDashWingedStrawberry.jl
module ShroomHelperOneDashWingedStrawberry

using ..Ahorn, Maple

@mapdef Entity "ShroomHelper/OneDashWingedStrawberry" OneDashWingedStrawberry(x::Integer, y::Integer, despawnFromSessionIfDashedTwiceInSameRoom::Bool=false)

const placements = Ahorn.PlacementDict(
    "One Dash Winged Strawberry (Shroom Helper)" => Ahorn.EntityPlacement(
        OneDashWingedStrawberry
    )
)

function Ahorn.selection(entity::OneDashWingedStrawberry)
    x, y = Ahorn.position(entity)
    return Ahorn.Rectangle(x - 7, y - 7, 14, 14)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::OneDashWingedStrawberry, room::Maple.Room)
    Ahorn.drawSprite(ctx, "collectables/ghostgoldberry/wings01", 0, 0)
end

end

File: Loenn/entities/one_dash_winged_strawberry.lua
local utils = require("utils")

local one_dash_winged_strawberry = {}

one_dash_winged_strawberry.name = "ShroomHelper/OneDashWingedStrawberry"
one_dash_winged_strawberry.depth = 0
one_dash_winged_strawberry.placements = {
    name = "one_dash_winged_strawberry",
    data = {
        despawnFromSessionIfDashedTwiceInSameRoom = false,
    }
}

one_dash_winged_strawberry.texture = "collectables/ghostgoldberry/wings01"

function one_dash_winged_strawberry.selection(room, entity)
    return utils.rectangle(entity.x - 7, entity.y - 7, 14, 14)
end

return one_dash_winged_strawberry